packages/terra-slide-panel/src/SlidePanel.jsx

```diff
@@ -85,33 +85,65 @@ const defaultProps = {
   panelSize: 'small',
 };
 
+/**
+ * This method ensures that aria-expanded attribute is allowed with the node role to avoid accessibility violations.
+ * More info on associated roles can be found here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#associated_roles
+ * @param {HTMLElement} node - a node to check
+ * @returns true if node's role is compatible with aria-expanded attribute
+ */
+const roleIsCompartible = (node) => {
+  const role = node.getAttribute('role');
+  return (
+    role === 'application'
+    || role === 'button'
+    || role === 'checkbox'
+    || role === 'combobox'
+    || role === 'gridcell'
+    || role === 'link'
+    || role === 'listbox'
+    || role === 'menuitem'
+    || role === 'row'
+    || role === 'rowheader'
+    || role === 'tab'
+    || role === 'treeitem'
+    || role === null
+    || role === undefined
+  );
```

Comment:
This looks like a lot of conditions that will always be sequentially checked. Can we optimize it using a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) and the [Set.has method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has)?

Reply:
And I'm confused why are we testing for all the possible values if we don't care that role is null/undefined?

Reply:
@sdadn aria-expanded attribute should be used [only with specific roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#associated_roles), otherwise it causes accessibility violation. This attribute was not introduced in this change, but it was causing accessibility violations with specific roles only, so I added a check. I'll double check with @mjpalazzo what would be the best way to address that